akka-http-core/src/main/scala/akka/http/impl/settings/ServerSentEventSettingsImpl.scala

```diff
@@ -0,0 +1,28 @@
+/*
+ * Copyright (C) 2009-2018 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package akka.http.impl.settings
+
+import akka.annotation.InternalApi
+import akka.http.impl.util.SettingsCompanion
+import com.typesafe.config.Config
+
+@InternalApi
+private[http] final case class ServerSentEventSettingsImpl(
+  maxEventSize: Int,
+  maxLineSize:  Int
+) extends akka.http.scaladsl.settings.ServerSentEventSettings {
+  require(maxLineSize > 0, "max-line-size must be greater than 0")
+  require(maxEventSize > maxLineSize, "max-event-size must be greater than max-line-size")
+
+  override def productPrefix: String = "ServerSentEventSettings"
+
+}
+
+object ServerSentEventSettingsImpl extends SettingsCompanion[ServerSentEventSettingsImpl]("akka.http.sse") {
+  def fromSubConfig(root: Config, c: Config) = ServerSentEventSettingsImpl(
+    c getInt "max-event-size",
```

Comment:
Do we really want to use infix notation?

Reply:
Oops, I did not know. Well, let's see what @ktoso or the other reviewers from the Akka team think. But anyway, great work!